dlib/serialize.h

```diff
@@ -2555,7 +2555,7 @@ namespace dlib
     }  
     
     #define DLIB_DEFINE_DEFAULT_SERIALIZATION(Type, ...)                \
-    void serialize_to(std::ostream& out) const                          \
+    void serialize_to(std::ostream& __out) const                        \
```

Comment:
if I remember correctly, identifiers starting with double underscore are reserved, and may cause ub.

Reply:
What convention do you prefer? ```out``` and ```in``` need to be refactored to something sensible that would likely not conflict with member variables of custom types.

Reply:
@davisking Might have an opinion on appropriate names.

Reply:
I would probably just use `out_` and `in_`, or `out_param_` and `in_param_` if you want to reduce the collision odds.

Reply:
I think i prefer ```out_ref_``` and ```in_ref_``` as private member variables are often either prefixed or suffixed with ```_```.

Reply:
I don't know if a silly name like ```out_avoid_conflict_with_member_variable_name``` is ok?

Reply:
I've seen @davisking use names like that when writing visual studio workarounds.

Reply:
Yeah, it is illegal to use names with double underscores.  But yeah, these names are not good.  It's also a problem that `version` is simple name too.  I would prefix in, out, and version with some crazy string.  Like turn `out` into `dlibDefaultSer$_out`.   No one will ever type `dlibDefaultSer$_`.